loky/backend/utils.py

```diff
@@ -1,7 +1,77 @@
+import os
+import sys
+import errno
+import signal
+import warnings
 import threading
+import subprocess
 
 
 def _flag_current_thread_clean_exit():
     """Put a ``_clean_exit`` flag on the current thread"""
     thread = threading.current_thread()
     thread._clean_exit = True
+
+
+def safe_terminate(process):
+    """Terminate a process and its children.
+    """
+    try:
+        _safe_terminate(process.pid)
+        process.join()
+    except OSError as e:
+        import traceback
+        tb = traceback.format_exc()
+        warnings.warn("Failure in child introspection on this platform. You "
+                      "should report it on https://github.com/tomMoral/loky "
+                      "with the following traceback\n{}".format(tb))
+        # In case we cannot introspect the children, we fall back to the
+        # classic Process.terminate.
+        process.terminate()
+        process.join()
+
+
+def _safe_terminate(pid):
+    """Terminate the children of a process before killing this process.
```

Comment:
Recursively kill the descendants of a process before killing the process itself.